### PR TITLE
Remove deleted messages from the unread message count

### PR DESCRIPTION
**Pull request by parasharrajat**
### Fixed Issues

$ https://github.com/Expensify/App/issues/5237

### Tests | QA Steps
1. Open the App on the web with account A.
2. Send a few messages. Count the number.
3. Delete some of the recently sent messages. Count the remaining number.
4. Log out, log in with account B for the same Chat.
5. Count should be correct for unread messages.

### Tested On

- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [ ] iOS
- [ ] Android

#### Web

https://user-images.githubusercontent.com/24370807/140608201-4fb6c060-8222-4e5a-afff-a141bd0cfc4d.mp4

### CODE REVIEW

**Comment by parasharrajat** [comment]
Awaiting review....

**Comment by puneetlath** [comment]
@parasharrajat a couple interesting things I noticed while testing this: 

1. When you mark a chat as unread, the unread count is immediately updated in the tab bar. However, when you delete a comment, the unread count in the tab bar in the current tab isn't updated. It is correct if you open in a new tab, but it doesn't update the existing tab. Why is that? 
2. When you open the app in the new tab, the tab bar shows the proper unread count, but the "new" line that indicates the point at which the messages are unread from isn't shown. Do you want to raise a new issue for this.

**Comment by parasharrajat** [comment]
>When you mark a chat as unread, the unread count is immediately updated in the tab bar. However, when you delete a comment, the unread count in the tab bar in the current tab isn't updated. It is correct if you open in a new tab, but it doesn't update the existing tab. Why is that?

I think because we manually set the count. And don't do that when a message is deleted. Do you want me to look into that as well in this PR?

> When you open the app in the new tab, the tab bar shows the proper unread count, but the "new" line that indicates the point at which the messages are unread from isn't shown. Do you want to raise a new issue for this?

I noticed this too. The new line indicator is broken again and it is not predictable. Sometimes shows and somtimes not. I was about to raise this as well after review.

**Comment by puneetlath** [comment]
>Do you want me to look into that as well in this PR?

Yes, I think we should do that as well. We can increase the payment by $250 to account for the increase in scope. cc @mallenexpensify 

>I was about to raise this as well after review.
Great!

**Comment by puneetlath** [comment]
The code generally looks good to me. I'll give it another review after that update. Thanks!

**Comment by parasharrajat** [comment]
Sure, Let me fix that.

**Comment by parasharrajat** [comment]
@puneetlath I have fixed it. Please review.

File: src/libs/actions/ReportActions.js [hunk 1]
```diff
@@ -67,7 +72,31 @@ function isReportMissingActions(reportID, maxKnownSequenceNumber) {
         || reportActionsMaxSequenceNumbers[reportID] < maxKnownSequenceNumber;
 }
 
+/**
+ * Get the count of deleted messages upto a sequence number of a report
+ * @param {Number|String} reportID
+ * @param {Number} lastSequenceNumber
```
Comment (puneetlath) [inline src/libs/actions/ReportActions.js:78]: Similarly, this isn't really necessarily the "last" sequence number, just any sequence number given, so I think a slightly clearer variable name would be just `sequenceNumber`.

**Review by puneetlath — DISMISSED** [review]

**Comment by puneetlath** [comment]
@parasharrajat I left you two minor comments, otherwise it looks good to me. Let me know what you think and then I'm happy to merge.

File: src/libs/actions/ReportActions.js [hunk 2]
```diff
@@ -67,7 +72,31 @@ function isReportMissingActions(reportID, maxKnownSequenceNumber) {
         || reportActionsMaxSequenceNumbers[reportID] < maxKnownSequenceNumber;
 }
 
+/**
+ * Get the count of deleted messages upto a sequence number of a report
```
[the remark this answers is not in the document]
Comment (parasharrajat) [reply src/libs/actions/ReportActions.js:76]: I think yes. We both are correct.
This function gives the count from latest to a number but not including that.

Let me know. I can swap it.

File: src/libs/actions/ReportActions.js (hunk 1 above)
Comment (parasharrajat) [reply src/libs/actions/ReportActions.js:78]: Correct. Updating ...

**Comment by parasharrajat** [comment]
@puneetlath Updated those. Thanks for pointing it out.

File: src/libs/actions/ReportActions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (puneetlath) [reply src/libs/actions/ReportActions.js]: Ah, yes I see what you mean. In any case, thanks for updating!

**Review by puneetlath — APPROVED** [review]

**Pull request merged**